sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -237,8 +237,10 @@ case class DropTableCommand(
 
     if (isTempView || catalog.tableExists(tableName)) {
       try {
+        val hasViewText = isTempView &&
+          catalog.getTempViewOrPermanentTableMetadata(tableName).viewText.isDefined
         sparkSession.sharedState.cacheManager.uncacheQuery(
-          sparkSession.table(tableName), cascade = !isTempView)
+          sparkSession.table(tableName), cascade = !isTempView || hasViewText)
```

Comment:
how about the v2 drop table command?

Reply:
Hmm, I think all dropping temp views are handled by v1 `DropTableCommand` and v2 only handles tables, is that correct?

Reply:
ah you are right.